components/raftstore/src/store/fsm/apply.rs

```diff
@@ -6693,4 +6690,97 @@ mod tests {
         rx.recv_timeout(Duration::from_millis(500)).unwrap();
         system.shutdown();
     }
+
+    fn new_batch_split_request(keys: Vec<Vec<u8>>) -> AdminRequest {
+        let mut req = AdminRequest::default();
+        req.set_cmd_type(AdminCmdType::BatchSplit);
+        for key in keys {
+            let mut split_req = SplitRequest::default();
+            split_req.set_split_key(key);
+            split_req.set_new_peer_ids(vec![1]);
+            req.mut_splits().mut_requests().push(split_req);
+        }
+        req
+    }
+
+    #[test]
+    fn test_validate_batch_split() {
+        let mut region = Region::default();
+        region.set_start_key(b"k05".to_vec());
+        region.set_end_key(b"k10".to_vec());
+        region.set_peers(vec![new_peer(1, 2)].into());
+
+        let fail_cases = vec![
+            // case: default admin request should be rejected
+            (AdminRequest::default(), "missing split requests"),
+            // case: split is deprecated
+            (
+                {
+                    let mut req = AdminRequest::default();
+                    req.set_cmd_type(AdminCmdType::Split);
+                    let mut split_req = SplitRequest::default();
+                    split_req.set_split_key(b"k06".to_vec());
+                    req.set_split(split_req);
+                    req
+                },
+                "missing split requests",
```

Comment:
all error messages should use variable as they're mostly duplicated.

Reply:
Done